### PR TITLE
Shrapnel mines no longer crash server, cleans up pellets a bit

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Groups of shrapnel mines no longer crash the server with a giant mess of recursion from bad trigger logic. Flying objects also no longer trigger mines while passing through, though sparks from nearby igniting mines and damaging them directly will still trigger them. Note flying things not setting off mines isn't the fix, that's just a side effect to neatening some landmine code.

You can set off a 5x5-ish square of shrapnel mines without killing the server though i obviously don't offer warranties on anything above that

Also I accidentally made security cargo packs require forensics access with the shrapnel PR, this fixes that.

## Why It's Good For The Game
Fixe

## Changelog
:cl: Ryll/Shaps
fix: Clusters of shrapnel mines will no longer kill the server when set off together.
tweak: Shooting over landmines will no longer trigger them
fix: security crates briefly required forensics access instead of proper security access, this has been fixed
/:cl:

### CODE REVIEW

File: code/game/objects/effects/mines.dm [hunk 1]
```diff
@@ -51,6 +59,7 @@
 	name = "shrapnel mine"
 	var/shrapnel_type = /obj/projectile/bullet/shrapnel
 	var/shrapnel_magnitude = 3
+	del_self = FALSE
```
Comment (AnturK) [inline code/game/objects/effects/mines.dm:62]: I don't understand why is this necessary. Please explain the issue and solution.

Comment (Ryll-Ryll) [reply code/game/objects/effects/mines.dm:62]: A pellet cloud is a component that is used to generate shrapnel, and is added to things like grenades and landmines. When the grenade or landmine goes off, the cloud component shoots a bunch of shrapnel everywhere. To cut down on spam for someone getting hit by multiple pellets, the pellet clouds track each pellet and what they hit, so when all the pellets have hit something or run out of range, we tell the person they're hit by 5 pellets instead of 5 separate messages.

Because tracking the pellets requires the component staying around after the grenade or landmine goes off, we nullspace the mine or grenade rather than qdel'ing it outright to allow the component to survive, then qdel the grenade/mine and component once the pellets are finished.

If you'd like I can add more documentation for my reasoning

Comment (AnturK) [reply code/game/objects/effects/mines.dm:62]: This should be handled by the component then, hook up to pre delete signal, prevent destruction. move to nullspace then delete after component is done.

**Comment by Ryll-Ryll** [comment]
Ugh I pushed some updates to this but I think it's severed because of the git forking, I'll deal with this soon

**Comment by Ryll-Ryll** [comment]
Closing due to git issue, will remake
